boards/arm/pico_pi_m4/pinmux.c

```diff
@@ -0,0 +1,187 @@
+
+
+#include <init.h>
+#include "device_imx.h"
+
+static int pico_pi_m4_pinmux_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+ #ifdef CONFIG_UART_IMX_UART_2
+	IOMUXC_SW_MUX_CTL_PAD_UART2_RX_DATA =
+		IOMUXC_SW_MUX_CTL_PAD_UART2_RX_DATA_MUX_MODE(0);
+	IOMUXC_SW_MUX_CTL_PAD_UART2_TX_DATA =
+		IOMUXC_SW_MUX_CTL_PAD_UART2_TX_DATA_MUX_MODE(0);
+	IOMUXC_SW_PAD_CTL_PAD_UART2_RX_DATA =
+		IOMUXC_SW_PAD_CTL_PAD_UART2_RX_DATA_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_UART2_RX_DATA_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_UART2_RX_DATA_HYS_MASK |
+		IOMUXC_SW_PAD_CTL_PAD_UART2_RX_DATA_DSE(0);
+
+	IOMUXC_SW_PAD_CTL_PAD_UART2_TX_DATA =
+		IOMUXC_SW_PAD_CTL_PAD_UART2_TX_DATA_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_UART2_TX_DATA_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_UART2_TX_DATA_HYS_MASK |
+		IOMUXC_SW_PAD_CTL_PAD_UART2_TX_DATA_DSE(0);
+
+	IOMUXC_UART2_RX_DATA_SELECT_INPUT =
+		IOMUXC_UART2_RX_DATA_SELECT_INPUT_DAISY(2);
+ #endif  /* CONFIG_UART_IMX_UART_2 */
+
+
+ #ifdef CONFIG_UART_IMX_UART_5
+	IOMUXC_SW_MUX_CTL_PAD_I2C4_SCL =
+		IOMUXC_SW_MUX_CTL_PAD_I2C4_SCL_MUX_MODE(1);
+	IOMUXC_SW_MUX_CTL_PAD_I2C4_SDA =
+		IOMUXC_SW_MUX_CTL_PAD_I2C4_SDA_MUX_MODE(1);
+
+	IOMUXC_SW_PAD_CTL_PAD_I2C4_SCL =
+		IOMUXC_SW_PAD_CTL_PAD_I2C4_SCL_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_I2C4_SCL_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_I2C4_SCL_HYS_MASK |
+		IOMUXC_SW_PAD_CTL_PAD_I2C4_SCL_DSE(0);
+
+	IOMUXC_SW_PAD_CTL_PAD_I2C4_SDA =
+		IOMUXC_SW_PAD_CTL_PAD_I2C4_SDA_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_I2C4_SDA_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_I2C4_SDA_HYS_MASK |
+		IOMUXC_SW_PAD_CTL_PAD_I2C4_SDA_DSE(0);
+
+	IOMUXC_UART5_RX_DATA_SELECT_INPUT =
+		IOMUXC_UART5_RX_DATA_SELECT_INPUT_DAISY(0);
+
+ #endif  /* CONFIG UART_IMX_UART_5 */
+
+
+ #ifdef CONFIG_UART_IMX_UART_6
+	IOMUXC_SW_MUX_CTL_PAD_EPDC_DATA08 =
+		IOMUXC_SW_MUX_CTL_PAD_EPDC_DATA08_MUX_MODE(3);
+	IOMUXC_SW_MUX_CTL_PAD_EPDC_DATA09 =
+		IOMUXC_SW_MUX_CTL_PAD_EPDC_DATA09_MUX_MODE(3);
+
+	IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA08 =
+		IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA08_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA08_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA08_HYS_MASK |
+		IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA08_DSE(1);
+
+	IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA09 =
+		IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA09_PE_MASK  |
+		IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA09_PS(3)    |
+		IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA09_HYS_MASK |
+		IOMUXC_SW_PAD_CTL_PAD_EPDC_DATA09_DSE(1);
+
+	IOMUXC_UART6_RX_DATA_SELECT_INPUT =
+		IOMUXC_UART6_RX_DATA_SELECT_INPUT_DAISY(0);
+ #endif  /* CONFIG_UART_IMX_UART_6 */
+
+
+ #ifdef CONFIG_I2C_1
+	IOMUXC_SW_MUX_CTL_PAD_UART1_RX_DATA =
+		IOMUXC_SW_MUX_CTL_PAD_UART1_RX_DATA_MUX_MODE(1) |
+		IOMUXC_SW_MUX_CTL_PAD_UART1_RX_DATA_SION_MASK;
+	IOMUXC_SW_MUX_CTL_PAD_UART1_TX_DATA =
+		IOMUXC_SW_MUX_CTL_PAD_UART1_TX_DATA_MUX_MODE(1) |
+		IOMUXC_SW_MUX_CTL_PAD_UART1_TX_DATA_SION_MASK;
+
+	IOMUXC_I2C1_SCL_SELECT_INPUT = IOMUXC_I2C1_SCL_SELECT_INPUT_DAISY(0);
+	IOMUXC_I2C1_SDA_SELECT_INPUT = IOMUXC_I2C1_SDA_SELECT_INPUT_DAISY(0);
+
+	IOMUXC_SW_PAD_CTL_PAD_UART1_RX_DATA =
```

Comment:
You seem to be mixing I2C and UART within this `#ifdef CONFIG_I2C_1` block. Is that intended?